nf_core/pipeline-template/.github/workflows/awsfulltest.yml

```diff
@@ -24,22 +22,24 @@ jobs:
     if: github.repository == '{{ name }}'
     runs-on: ubuntu-latest
     steps:
-      - name: Setup Miniconda
-        uses: conda-incubator/setup-miniconda@v2
-        with:
-          auto-update-conda: true
-          python-version: 3.7
-      - name: Install awscli
-        run: conda install -c conda-forge awscli
-      - name: Start AWS batch job
+      - name: Launch workflow via tower
         # TODO nf-core: You can customise AWS full pipeline tests as required
         # Add full size test data (but still relatively small datasets for few samples)
         # on the `test_full.config` test runs with only one set of parameters
-        # Then specify `-profile test_full` instead of `-profile test` on the AWS batch command
         run: |
-          aws batch submit-job \
-            --region eu-west-1 \
-            --job-name nf-core-{{ short_name }} \
-            --job-queue $AWS_JOB_QUEUE \
-            --job-definition $AWS_JOB_DEFINITION \
-            --container-overrides '{"command": ["{{ name }}", "-r '"${GITHUB_SHA}"' -profile test --outdir s3://'"${AWS_S3_BUCKET}"'/{{ short_name }}/results-'"${GITHUB_SHA}"' -w s3://'"${AWS_S3_BUCKET}"'/{{ short_name }}/work-'"${GITHUB_SHA}"' -with-tower"], "environment": [{"name": "TOWER_ACCESS_TOKEN", "value": "'"$TOWER_ACCESS_TOKEN"'"}]}'
+          curl -X POST "https://api.tower.nf/workflow/launch?workspaceId=${TOWER_WORKSPACE_ID}" \
+          -H "Accept: application/json" \
+          -H "Authorization: Bearer ${TOWER_BEARER_TOKEN}" \
+          -H "Content-Type: application/json" \
+          -d '{
+            "launch": {
+              "computeEnvId" : "'"${TOWER_COMPUTE_ENV}"'",
+              "pipeline" : "{{ name }}",
+              "workDir" : "s3://'"${AWS_S3_BUCKET}"'/work/{{ short_name }}/commit-'"${GITHUB_SHA}"'",
+              "paramsText" : "{\"outdir\":\"s3://'"${AWS_S3_BUCKET}"'/{{ short_name }}/results-'"${GITHUB_SHA}"'\"}",
+              "revision" : "'"${GITHUB_SHA}"'",
```

Comment:
How come some of these are quoted with `"'"` and some are not?

Reply:
Ah it's because env variables don't get into single-quoted strings, but you need that to avoid having to escape every `"`. Ok, lesser of evils maybe!

Reply:
yeah I played with different options on how to do proper escaping and variable setting, and that's the only option that was working for me